[PL-52150] SMP 0.14.11 Patch RN (#7237)
harness/developer-hub: release-notes/self-managed-enterprise-edition.md

:::

> ## June 29, 2024, hotfix 0.14.6

Reviewer: It should be 0.14.7

Reviewer 2: Its 0.14.11, last HF release was 0.14.10

Author: @avaakash That's what I thought too, but I asked in the channel, and it was confirmed as 0.14.6.
@bharatgoelharness @ankittiwari-harness please confirm

Author: https://harness.slack.com/archives/C07AQB8M3CZ/p1719655288659129?thread_ts=1719653409.728879&cid=C07AQB8M3CZ

> ## June 29, 2024, hotfix 0.14.6

Reviewer 2: ```suggestion
## June 29, 2024, hotfix 0.14.11
```
This will be going with 0.14.11 SMP HF version

> - Pipeline steps were stuck because of an issue with `MaxConcurrentChildCallback` when the max concurrency was configured with a high maximum amount of steps. Harness upgraded the `pipeline-service` to 1.61.0 to resolve this issue. (PIPE-20146, ZD-66018)

Reviewer 3: we upgraded to 1.61.7

> - Pipeline steps were stuck because of an issue with `MaxConcurrentChildCallback` when the max concurrency was configured with a high maximum amount of steps. Harness upgraded the `pipeline-service` to 1.61.0 to resolve this issue. (PIPE-20146, ZD-66018)

Reviewer 2: ```suggestion
- Pipeline steps were stuck because of an issue with `MaxConcurrentChildCallback` when the max concurrency was configured with a high maximum amount of steps. Harness upgraded the `pipeline-service` to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)
```
The fix is in pipeline service version 1.61.7

> - Pipeline steps were stuck because of an issue with `MaxConcurrentChildCallback` when the max concurrency was configured with a high maximum amount of steps. Harness upgraded the `pipeline-service` to 1.61.0 to resolve this issue. (PIPE-20146, ZD-66018)

Reviewer: Upgraded pipeline service from 1.61.5 to 1.61.7

> - Pipeline steps were stuck because of an issue with `MaxConcurrentChildCallback` when the max concurrency was configured with a high maximum amount of steps. Harness upgraded the `pipeline-service` to 1.61.0 to resolve this issue. (PIPE-20146, ZD-66018)

Reviewer 3: Customer might not be aware with MaxConcurrentChildCallback, We can mention,

Pipelines were getting stuck when the pipeline was executing steps at max concurrency intermittently, we identified a fix for the same

> - Pipeline steps were stuck because of an issue with `MaxConcurrentChildCallback` when the max concurrency was configured with a high maximum amount of steps. Harness upgraded the `pipeline-service` to 1.61.0 to resolve this issue. (PIPE-20146, ZD-66018)

Author: ```suggestion
- Pipelines were getting stuck when the pipeline was executing steps at max concurrency intermittently. Harness identified the fix and upgraded the `pipeline-service` from 1.61.5 to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)
```

   Harness recommends that you increase your Redis memory to a minimum of 10GB to prevent future potential issues.

> 

Author: ```suggestion
```

>    ```yaml

Reviewer: This is only specific to customers using a lot of concurrent deployments, not specific to every customer

| NG Manager | 1.24.7 |
| CI Manager | 1.12.5 |
| Pipeline Service | 1.61.7 |
| Platform Service | 1.12.0 |
| Access Control Service | 1.33.2 |
| Delegate | 24.02.82203 |
| Change Data Capture | 1.5.3 |
| STO Core | 1.83.8 |
| Test Intelligence Service | 1.8.1 |
| NG UI | 1.7.6 |
| LE NG | 1.1.0 |

**Alternative air gap bundle download method**

Some admins might not have Google account access to download air gap bundles. As an alternative, you can use `gsutil`. For `gsutil` installation instructions, go to [Install gsutil](https://cloud.google.com/storage/docs/gsutil_install) in the Google Cloud documentation.

```
gsutil -m cp \
  "gs://smp-airgap-bundles/harness-0.14.11/ccm_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/cdng_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ce_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/cet_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ci_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ff_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/platform_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/sto_images.tgz" \
  .
```

#### Fixed issues

> - Pipelines were getting stuck when the pipeline was executing steps at max concurrency intermittently. This issue only affected customers with a large number of concurrent deployments. Harness identified the fix and upgraded the `pipeline-service` from 1.61.5 to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)

Author: ```suggestion
- Pipelines were getting stuck intermittently when the pipeline was executing steps at max concurrency. This issue only affected customers with a large number of concurrent deployments. Harness identified the fix and upgraded the `pipeline-service` from 1.61.5 to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)
```

| NG Manager | 1.24.7 |
| CI Manager | 1.12.5 |
| Pipeline Service | 1.61.7 |
| Platform Service | 1.12.0 |
| Access Control Service | 1.33.2 |
| Delegate | 24.02.82203 |
| Change Data Capture | 1.5.3 |
| STO Core | 1.83.8 |
| Test Intelligence Service | 1.8.1 |
| NG UI | 1.7.6 |
| LE NG | 1.1.0 |

**Alternative air gap bundle download method**

Some admins might not have Google account access to download air gap bundles. As an alternative, you can use `gsutil`. For `gsutil` installation instructions, go to [Install gsutil](https://cloud.google.com/storage/docs/gsutil_install) in the Google Cloud documentation.

```
gsutil -m cp \
  "gs://smp-airgap-bundles/harness-0.14.11/ccm_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/cdng_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ce_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/cet_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ci_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ff_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/platform_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/sto_images.tgz" \
  .
```

#### Fixed issues

> - Pipelines were getting stuck when the pipeline was executing steps at max concurrency intermittently.  This issue only affected customers with a large number of concurrent deployments. Harness identified the fix and upgraded the `pipeline-service` from 1.61.5 to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)
> 

Author: ```suggestion
- Pipelines were getting stuck when the pipeline was executing steps at max concurrency intermittently. This issue only affected customers with a large number of concurrent deployments. Harness identified the fix and upgraded the `pipeline-service` from 1.61.5 to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)
```

| NG Manager | 1.24.7 |
| CI Manager | 1.12.5 |
| Pipeline Service | 1.61.7 |
| Platform Service | 1.12.0 |
| Access Control Service | 1.33.2 |
| Delegate | 24.02.82203 |
| Change Data Capture | 1.5.3 |
| STO Core | 1.83.8 |
| Test Intelligence Service | 1.8.1 |
| NG UI | 1.7.6 |
| LE NG | 1.1.0 |

**Alternative air gap bundle download method**

Some admins might not have Google account access to download air gap bundles. As an alternative, you can use `gsutil`. For `gsutil` installation instructions, go to [Install gsutil](https://cloud.google.com/storage/docs/gsutil_install) in the Google Cloud documentation.

```
gsutil -m cp \
  "gs://smp-airgap-bundles/harness-0.14.11/ccm_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/cdng_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ce_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/cet_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ci_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/ff_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/platform_images.tgz" \
  "gs://smp-airgap-bundles/harness-0.14.11/sto_images.tgz" \
  .
```

#### Fixed issues

> - - Pipelines were getting stuck when the pipeline was executing steps at max concurrency intermittently.  This issue only affected customers with a large number of concurrent deployments. Harness identified the fix and upgraded the `pipeline-service` from 1.61.5 to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)

Author: ```suggestion
- Pipelines were getting stuck when the pipeline was executing steps at max concurrency intermittently.  This issue only affected customers with a large number of concurrent deployments. Harness identified the fix and upgraded the `pipeline-service` from 1.61.5 to 1.61.7 to resolve this issue. (PIPE-20146, ZD-66018)
```

   Harness recommends that you increase your Redis memory to a minimum of 10GB to prevent future potential issues.

   ```yaml
   platform:
     bootstrap:
       database:
         redis:
           redis:
             resources:
               limits:
                 memory: 10Gi
               requests:
                 cpu: 1
                 memory: 10Gi

>    ```

Reviewer 3: Can we remove this from RN. This is specific to openAI

>    Harness recommends that you increase your Redis memory to a minimum of 10GB to prevent future potential issues.
> 
> 
>    ```yaml
>    platform:
>      bootstrap:
>        database:
>          redis:
>            redis:
>              resources:
>                limits:
>                  memory: 10Gi
>                requests:
>                  cpu: 1
>                  memory: 10Gi
>    ```

Author: ```suggestion
```
